state/state_test.go

```diff
@@ -5,74 +5,87 @@ import (
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+
 	abci "github.com/tendermint/abci/types"
+
 	crypto "github.com/tendermint/go-crypto"
-	cfg "github.com/tendermint/tendermint/config"
+
 	dbm "github.com/tendermint/tmlibs/db"
 	"github.com/tendermint/tmlibs/log"
+
+	cfg "github.com/tendermint/tendermint/config"
 )
 
-func TestStateCopyEquals(t *testing.T) {
+// setupTestCase does setup common to all test cases
+func setupTestCase(t *testing.T) (func(t *testing.T), dbm.DB, *State) {
 	config := cfg.ResetTestRoot("state_")
-
-	// Get State db
 	stateDB := dbm.NewDB("state", config.DBBackend, config.DBDir())
 	state := GetState(stateDB, config.GenesisFile())
 	state.SetLogger(log.TestingLogger())
 
+	tearDown := func(t *testing.T) {}
```

Comment:
why bother with this?

Reply:
In case we are going to add some teardown behaviour in the future. I wanted to make it obvious how to add it.